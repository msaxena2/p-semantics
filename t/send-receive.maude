in "../p-semantics.maude"

mod P-SEND-RECIEVE-TEST is including P-SEMANTICS .

  ops Sender Receiver Init SendRequests Done ePing receiver : -> Id .

  op Receiver : -> Type .
  ops SenderPgm ReceiverPgm : -> Stmt .

  eq SenderPgm = (
    machine Sender sends ePing ; {
      var receiver : Receiver ;

      start state Init {
        entry {
          receiver = new Receiver( .Exps );
          goto SendRequests ;
        }
      }

      state SendRequests {
        entry {
          send receiver, ePing ;
        }
      }
    } ) .

   eq ReceiverPgm = (
    machine Receiver ; {
      start state Init {
        on ePing do {
          goto Done ;
        }
      }
    } ) .

endm

red SenderPgm .
red ReceiverPgm .

red init(SenderPgm ReceiverPgm) .

rew init(SenderPgm ReceiverPgm) .

quit .
