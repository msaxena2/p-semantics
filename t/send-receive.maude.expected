==========================================
reduce in P-SEND-RECIEVE-TEST : SenderPgm .
result MachineStmt: machine Sender ; {
  var receiver : Receiver ;
  start state Init {
    entry {
      receiver = new Receiver(.Exps) ;
      goto SendRequests ;
    }
  }
  state SendRequests {
    entry {
      send receiver,ePing ;
    }
  }
}
==========================================
reduce in P-SEND-RECIEVE-TEST : ReceiverPgm .
result MachineStmt: machine Receiver ; {
  start state Init {
    on ePing do {
      goto Done ;
    }
  }
}
==========================================
reduce in P-SEND-RECIEVE-TEST : init(SenderPgm
ReceiverPgm) .
result [Cfg]: 
<machines>  .MachineSchemas
</machines> 
<k>  machine Sender ; {
    var receiver : Receiver ;
    start state Init {
      entry {
        receiver = new Receiver(.Exps) ;
        goto SendRequests ;
      }
    }
    state SendRequests {
      entry {
        send receiver,ePing ;
      }
    }
  }
  machine Receiver ; {
    start state Init {
      on ePing do {
        goto Done ;
      }
    }
  }
</k> 
<soup>  
.Cell
</soup>
==========================================
rewrite in P-SEND-RECIEVE-TEST : init(SenderPgm
ReceiverPgm) .
result [Cfg]: 
<machines>  (machine Sender ; {
    var receiver : Receiver ;
    start state Init {
      entry {
        receiver = new Receiver(.Exps) ;
        goto SendRequests ;
      }
    }
    state SendRequests {
      entry {
        send receiver,ePing ;
      }
    }
  }) ;; machine Receiver ; {
    start state Init {
      on ePing do {
        goto Done ;
      }
    }
  }
</machines> 
<k>  .Stmt
</k> 
<soup>  
.Cell
</soup>
Bye.
